protocols/kad/src/behaviour/test.rs

```diff
@@ -1378,6 +1378,89 @@ fn network_behaviour_on_address_change() {
     );
 }
 
+#[test]
+fn network_behaviour_on_new_external_address_of_peer_for_present_entry() {
```

Comment:
If we delegate the event to `Kademlia::add_address`, I don't think we need a dedicated test for it!